File: examples/governance/model_registry_domain/lib/model_registry_domain/ports/data_usage_agreement_repository.rb
module ModelRegistryDomain
  module Ports
    module DataUsageAgreementRepository
      def find(id)
        raise NotImplementedError, "#{self.class}#find not implemented"
      end

      def save(data_usage_agreement)
        raise NotImplementedError, "#{self.class}#save not implemented"
      end

      def delete(id)
        raise NotImplementedError, "#{self.class}#delete not implemented"
      end
    end
  end
end

File: examples/governance_static_go/model_registry_static_go/domain/data_usage_agreement_repository.go
package domain

type DataUsageAgreementRepository interface {
	Find(id string) (*DataUsageAgreement, error)
	Save(DataUsageAgreement *DataUsageAgreement) error
	All() ([]*DataUsageAgreement, error)
	Delete(id string) error
	Count() (int, error)
}
